### ISSUE TITLE
Getting errors on create awx_inventory_source

### ISSUE THREAD

**Issue by fernandoeickhoff-lk** [denouche/terraform-provider-awx#33]
I am trying to configure an `awx_inventory_source` but I am not having success with it.

This is the error that I get:
```
╷
│ Error: Unable to create Inventory Source
│ 
│   with awx_inventory_source.use1_labs_eks_worker,
│   on main.tf line 23, in resource "awx_inventory_source" "use1_labs_eks_worker":
│   23: resource "awx_inventory_source" "use1_labs_eks_worker" {
│ 
│ Unable to create Inventory Source got Errors:
│ - detail: []
╵
Releasing state lock. This may take a few moments...
ERRO[0031] Hit multiple errors:
Hit multiple errors:
exit status 1 
```

Below is my `awx_inventory` and `awx_inventory_source` definition:

```
resource "awx_inventory" "use1_labs_eks_worker" {
  name            = "use1-labs-eks-worker"
  description     = "eks-worker inventory in use1 and labs"
  organization_id = awx_organization.default.id
}
```

```
resource "awx_inventory_source" "use1_labs_eks_worker" {
  name              = "use1-labs-eks-worker"
  description       = "all eks-worker instances in labs and use-1 region"
  credential_id     = data.awx_credential.aws.id
  inventory_id      = data.awx_inventory.use1_labs_eks_worker.id
  source            = "Amazon EC2"
  overwrite         = true
  update_on_launch  = true
  verbosity         = 1
  source_vars       = <<YAML
---
regions:
  - "us-east-1"
iam_role_arn: "arn:aws:iam::451476197489:role/terraform-ansible"
hostnames:
  - name: 'private-ip-address'
    separator: '_'
    prefix: 'tag:Name'
filters:
  tag:role: "eks_worker"
  tag:environment: "labs"
compose:
  ansible_host: private_ip_address
YAML
}
```

Thanks in advance!

**Comment by nickscott77** [comment]
Hi,

same issue here. Pls fix it. 
AWX Version: 21.12.0

**Comment by fernandoeickhoff-lk** [comment]
Still having this issue after such a long time without any kind of answer.

Figured out by myself the issue probably is related to this fix here: https://github.com/denouche/terraform-provider-awx/pull/20

Basically it seems that now it is mandatory to set the `source_project_id` in the `awx_inventory_source` even if the inventory `kind` is not a `scm` inventory. The `source_project_id` should only be mandatory for `scm` inventories.

Well, my inventory is not of scm kind, it is a regular one, empty string. 

 for_each = local.awx_inventories.vault

```
resource "awx_inventory" "awx_inventory_vault" {
name            = each.value.name
description     = each.value.description
kind            = ""
```

If I set an aleatory but valid project ID in the `inventory_source` it does not work:

```
resource "awx_inventory_source" "vault_sources" {
  for_each = local.inventory_sources

  inventory_id = awx_inventory.awx_inventory_vault[each.key].id
  name = each.value.source_name
  credential_id = data.awx_credential.aws.id
  description = each.value.source_description
  overwrite = true
  update_on_launch = true
  overwrite_vars = true
  source = "ec2"
  source_project_id = awx_project.awx_projects["master"].id
```

```
│ 
│ Unable to create Inventory Source got Errors:
│ - detail: [Cannot set source_project if not SCM type.]
╵
```

And if I don't set a project id it does not allow too:

`# source_project_id = awx_project.awx_projects["master"].id`

```
╷
│ Error: Unable to create Inventory Source
│ 
│   with awx_inventory_source.vault_sources["qa"],
│   on main.tf line 474, in resource "awx_inventory_source" "vault_sources":
│  474: resource "awx_inventory_source" "vault_sources" {
│ 
│ Unable to create Inventory Source got Errors:
│ - detail: []
╵
╷
│ Error: Unable to update Inventory Source
│ 
│   with awx_inventory_source.vault_sources["staging"],
│   on main.tf line 474, in resource "awx_inventory_source" "vault_sources":
│  474: resource "awx_inventory_source" "vault_sources" {
│ 
│ Unable to update Inventory Source with id 77: got Errors:
│ - source_project: [Invalid pk "0" - object does not exist.]
```

**Comment by fernandoeickhoff-lk** [comment]
Looks like this PR https://github.com/denouche/terraform-provider-awx/pull/29 mentioned in this issue https://github.com/denouche/terraform-provider-awx/issues/19 will solve this error but for some reason the PR was not merged yet

**Comment by jnahelou** [comment]
This issue was fixed by version 0.24.3.
It works using type scm:
```hcl
 # awx_inventory_source.vault_sources will be created                                                                                                                                                                                      
  + resource "awx_inventory_source" "vault_sources" {
      + description          = "demo"
      + id                   = (known after apply)
      + inventory_id         = 106
      + name                 = "demo"
      + overwrite            = true
      + overwrite_vars       = true
      + source               = "scm"
      + source_project_id    = 95
      + update_cache_timeout = 30
      + update_on_launch     = true
      + verbosity            = 1
    }
awx_inventory_source.vault_sources: Creating...
awx_inventory_source.vault_sources: Creation complete after 1s [id=810]
```

Or using EC2 
```hcl
 # awx_inventory_source.vault_sources will be created
  + resource "awx_inventory_source" "vault_sources" {
      + description          = "demo"
      + id                   = (known after apply)
      + inventory_id         = 106
      + name                 = "demo"
      + overwrite            = true
      + overwrite_vars       = true
      + source               = "ec2"
      + update_cache_timeout = 30
      + update_on_launch     = true
      + verbosity            = 1
    }

awx_inventory_source.vault_sources: Creating...
awx_inventory_source.vault_sources: Creation complete after 0s [id=812]
```
